kratos/tests/test_factories.py

```diff
@@ -16,6 +54,196 @@ def test_ExplicitBuilder(self):
+        settings = KM.Parameters("""
+        {
+            "name" : "newmark_scheme"
+        }
+        """)
+        self._auxiliary_test_function_Scheme(settings, "ResidualBasedNewmarkDisplacementScheme")
+
+    def test_ResidualBasedPseudoStaticDisplacementScheme(self):
+        settings = KM.Parameters("""
+        {
+            "name" : "pseudo_static_scheme", "rayleigh_beta_variable" : "PRESSURE"
+        }
+        """)
+        self._auxiliary_test_function_Scheme(settings, "ResidualBasedPseudoStaticDisplacementScheme")
+
+    def test_ResidualBasedBDFDisplacementScheme(self):
+        settings = KM.Parameters("""
+        {
+            "name" : "bdf_displacement_scheme"
+        }
+        """)
+        self._auxiliary_test_function_Scheme(settings, "ResidualBasedBDFDisplacementScheme")
+
+    def test_ResidualBasedBDFCustomScheme(self):
+        settings = KM.Parameters("""
+        {
+            "name" : "bdf_scheme"
+        }
+        """)
+        self._auxiliary_test_function_Scheme(settings, "ResidualBasedBDFCustomScheme")
+
+    def _auxiliary_test_function_Strategy(self, settings, name):
+        this_model = KM.Model()
+        model_part = this_model.CreateModelPart("Main")
+        strategy = KM.StrategyFactory().Create(model_part, settings)
+        self.assertTrue(KM.StrategyFactory().Has(settings["name"].GetString()))
+        self.assertEqual(strategy.Info(), name)
+
+    def test_ResidualBasedLinearStrategy(self):
+        settings = KM.Parameters("""
+        {
+            "name" : "linear_strategy",
+            "linear_solver_settings" : {
+                "solver_type" : "amgcl"
+            },
+            "scheme_settings" : {
+                "name" : "static_scheme"
+            },
+            "builder_and_solver_settings" : {
+                "name" : "elimination_builder_and_solver"
+            }
+        }
+        """)
+        self._auxiliary_test_function_Strategy(settings, "ResidualBasedLinearStrategy")
+
+    def test_ResidualBasedNewtonRaphsonStrategy(self):
+        settings = KM.Parameters("""
+        {
+            "name" : "newton_raphson_strategy",
+            "linear_solver_settings" : {
+                "solver_type" : "amgcl"
+            },
+            "scheme_settings" : {
+                "name" : "static_scheme"
+            },
+            "convergence_criteria_settings" : {
+                "name" : "displacement_criteria"
+            },
+            "builder_and_solver_settings" : {
+                "name" : "elimination_builder_and_solver"
+            }
+        }
+        """)
+        self._auxiliary_test_function_Strategy(settings, "ResidualBasedNewtonRaphsonStrategy")
+
+    def test_AdaptiveResidualBasedNewtonRaphsonStrategy(self):
+        settings = KM.Parameters("""
+        {
+            "name" : "adaptative_newton_raphson_strategy",
+            "linear_solver_settings" : {
+                "solver_type" : "amgcl"
```

Comment:
updating this will also be a bit of a pain
=> `solver_type` to `name`

Reply:
It is decided 100% that it is like that?

Reply:
yes - by you :)
https://github.com/KratosMultiphysics/Kratos/blob/a4d8f0cfa1c11d07a1bcc0e5b2e87f3d3be61a0b/kratos/factories/factory.h#L164

Reply:
Yes, but LinearSolverFactories work with solver_type, and in this PR I am not changing that. Can be done in another PR